### PR TITLE
Rewrite gulpfile and add svg task

### PR DESCRIPTION
**Pull request by Tschrock**
1. Moves the compilation/minify transforms to separate files and rewrites them with some better error handling
2. Adds an SVG task that runs using inkscape (>=1.0)
3. Modifies the watch task to only compile files that have changed
4. Adds fancy descriptions for tasks (visible using `gulp -T`)
5. Updates package.json with contributors and repo links

Hopefully I didn't break anything

Update:

6. Added a `.editorconfig` so Byte can stop complaining about formatting issues

### CODE REVIEW

File: tools/InkscapeIsDumbTransform.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+const PNG_START_SIGNATURE = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
+const PNG_END_SIGNATURE = Buffer.from([73, 69, 78, 68, 239, 191, 189, 66, 96, 239, 191, 189]);
+
+/**
+ * Filters out things that aren't a PNG because inkscape is dumb.
+ */
+exports.InkscapeIsDumbTransform = class extends Transform {
+
+	constructor() {
+        super();
```
Comment (BytewaveMLP) [inline tools/InkscapeIsDumbTransform.js]: lint: mixed tabs and spaces (rest of the project uses `\t`)

File: tools/InkscapeIsDumbTransform.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+const { Transform } = require('stream');
+
+const PNG_START_SIGNATURE = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
+const PNG_END_SIGNATURE = Buffer.from([73, 69, 78, 68, 239, 191, 189, 66, 96, 239, 191, 189]);
+
+/**
+ * Filters out things that aren't a PNG because inkscape is dumb.
+ */
+exports.InkscapeIsDumbTransform = class extends Transform {
```
Comment (BytewaveMLP) [inline tools/InkscapeIsDumbTransform.js]: Inconsistent `export` style

File: tools/InkscapeIsDumbTransform.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+		}
+		else {
+			// We either haven't started or already ended, so ignore the chunk
+		}
+
+		// Done processing this chunk
+		next();
+	}
+}
```
Comment (BytewaveMLP) [inline tools/InkscapeIsDumbTransform.js]: No newline at end of files

File: gulpfile.js [hunk 4]
```diff
@@ -1,93 +1,79 @@
...
 
-		let stdout = '';
-		let stderr = '';
+/**
+ * Cleans the build directories.
+ */
+function clean() {
+	return del(['gamemodes/**/*.lua']);
+}
+clean.description = "Cleans the build directories.";
```
Comment (BytewaveMLP) [inline gulpfile.js:14]: Mixed quotes

Comment (Tschrock) [reply gulpfile.js:14]: My quotes are mixed just the way I like them :v

**Review by NotMyWing — APPROVED** [review]

**Pull request merged**